lib/libc/minimal/source/string/string.c

```diff
@@ -59,6 +59,24 @@ char *strncpy(char *_MLIBC_RESTRICT d, const char *_MLIBC_RESTRICT s, size_t n)
 	return dest;
 }
 
+/**
+ *
+ * @brief Duplicate a string
+ *
+ * @return pointer to new string, or NULL if insufficient memory available
+ */
+
+char *strdup(const char *s)
+{
+	char *dest = malloc(strlen(s)+1);
+
+	if (dest) {
+		strcpy(dest, s);
```

Comment:
I fear this might trigger automatic lintian checks which discourage the use of known-unsafe functions like strcpy(), gets(), etc

Would be better to just save `strlen(s) + 1` into a local variable and then pass that to malloc() and then memcpy() instead of strcpy().

Reply:
Given a duff string and associated length, both memcpy and strcpy are going to be equally unsafe and using strcpy to me sort of makes more sense in a string handling function, but can change if you wish. @andrewboie ?

Reply:
@andrewboie Realised this is just stuck, are you wanting me to update to use strlen as suggested ?